synapse/storage/databases/main/roommember.py

```diff
@@ -131,10 +131,10 @@ def _transact(txn: LoggingTransaction) -> int:
                 query = """
                     SELECT COUNT(DISTINCT substr(out.user_id, pos+1))
                     FROM (
-                        SELECT rm.user_id as user_id, instr(rm.user_id, ':')
-                            AS pos FROM room_memberships as rm
-                        INNER JOIN current_state_events as c ON rm.event_id = c.event_id
-                        WHERE c.type = 'm.room.member'
```

Comment:
If https://github.com/matrix-org/synapse/pull/15734#discussion_r1220747946 is valid, then I think this simplification is also valid.

Reply:
Not a valid change, see https://github.com/matrix-org/synapse/pull/15734#discussion_r1220775214